libcaf_test/caf/test/fixture/flow.hpp

```diff
@@ -144,6 +144,81 @@ class CAF_TEST_EXPORT flow {
     caf::flow::coordinator* parent_;
   };
 
+  /// A subscription implementation without internal logic.
+  class passive_subscription_impl final
```

Comment:
Didn't we have something like this recently and used `never` instead?

Reply:
Yes, I considered using never_sub, but as it does not have the `demand` property used in https://github.com/actor-framework/actor-framework/pull/1696/files#diff-43095856bdeef9fbcd0ae3a4b69dc300d8420f533176e1c93c0733fcc9be9febR220-R221 I had to add it in the flow.

Reply:
Is that the only two places where do that check? The entire test is weird, since it says something about "the observer receives the item" but then the only inputs are `never`. Just for this one (weird) test, I wouldn't add this class.